cmd/otel-agent/config/agent_config_test.go

```diff
@@ -125,6 +126,34 @@ func (suite *ConfigTestSuite) TestAgentConfigWithDatadogYamlKeysAvailable() {
 	assert.Equal(t, 60, c.GetInt("agent_ipc.config_refresh_interval"))
 }
 
+func (suite *ConfigTestSuite) TestAgentConfigSetAPMFeaturesFromDatadogYaml() {
+	t := suite.T()
+	fileName := "testdata/config_default.yaml"
+	ddFileName := "testdata/datadog_apm_config_features.yaml"
+	c, err := NewConfigComponent(context.Background(), ddFileName, []string{fileName})
+	if err != nil {
+		t.Errorf("Failed to load agent config: %v", err)
+	}
+
+	assert.Equal(t, []string{"test1", "test2"}, c.GetStringSlice("apm_config.features"))
+}
+
+func (suite *ConfigTestSuite) TestAgentConfigSetAPMFeaturesFromEnv() {
+	t := suite.T()
+	fileName := "testdata/config_default.yaml"
+	oldval, _ := os.LookupEnv("DD_APM_FEATURES")
+	t.Setenv("DD_APM_FEATURES", "test1,test2")
+	defer func() {
+		t.Setenv("DD_APM_FEATURES", oldval)
+	}()
```

Comment:
You don't need the defer etc. `t.Setenv` is local and reset the env at the end of the test